src/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  time: Date,
+};
+
+export class Clock extends Component<Props, State> {
+  state: State = {
+    time: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ time: new Date() });
+      // eslint-disable-next-line
+      console.info(this.state.time.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { clockName } = prevProps;
+
+    if (clockName !== this.props.clockName) {
+      // eslint-disable-next-line
+      console.debug(`Renamed from ${clockName} to ${this.props.clockName}`)
```

Comment:
```suggestion
    const { clockName } = prevProps;
    const { clockName } = this.props;

    if (clockName !== clockName) {
      // eslint-disable-next-line
      console.debug(`Renamed from ${clockName} to ${clockName}`)
```

Reply:
In my opinion this.props.clockName more readable them clockName !== clockName , because it's so easy to get confused if this destructuring is much higher in the code)